tensorflow/python/training/server_lib.py

```diff
@@ -150,17 +150,22 @@ def __init__(self,
       self.start()
 
   def __del__(self):
+    # At shutdown, `errors` may have been garbage collected.
+    if errors is not None:
+      exception = errors.UnimplementedError
+    else:
+      exception = Exception
     try:
       c_api.TF_ServerStop(self._server)
       # Clean shutdown of servers is not yet implemented, so
       # we leak instead of calling c_api.TF_DeleteServer here.
       # See:
       # https://github.com/tensorflow/tensorflow/blob/0495317a6e9dd4cac577b9d5cf9525e62b571018/tensorflow/core/distributed_runtime/rpc/grpc_server_lib.h#L73
-    except errors.UnimplementedError:
-      pass
     except AttributeError:
       # At shutdown, `c_api` may have been garbage collected.
       pass
+    except exception:
```

Comment:
This seems dangerous, as we might be ignoring an important exception. Is there another way to go about this?

Reply:
I used the `Exception` directly because `errors.UnimplementedError` is derived from it. Maybe we can print the exception? I'm not sure if there is a good way to catch exceptions that occur in the `except xxx:` statement...

Reply:
Okay this is better, since if exceptions exists we are at least taking the option of ignoring an UnimplementedError.